src/kprofile.cpp

```diff
@@ -19,15 +25,43 @@ KProfile::KProfile(QObject *parent) :
 }
 
 #if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
-void KProfile::startDownload(QWebEngineDownloadItem* download)
+void KProfile::openFile(QWebEngineDownloadItem* download)
 #else
-void KProfile::startDownload(QWebEngineDownloadRequest* download)
+void KProfile::openFile(QWebEngineDownloadRequest* download)
 #endif
 {
     QString defaultFileName = download->url().fileName();
-    QString fileName = QFileDialog::getSaveFileName(KiwixApp::instance()->getMainWindow(),
-                                                       gt("save-file-as-window-title"), defaultFileName);
+    QTemporaryFile *tempFile = new QTemporaryFile(QDir::tempPath() + "/XXXXXX." + QFileInfo(defaultFileName).suffix());
+    if (tempFile->open()) {
+        QString tempFilePath = tempFile->fileName();
+        tempFile->close();
+#if QT_VERSION < QT_VERSION_CHECK(5, 15, 0)
+        download->setPath(tempFilePath);
+#else
+        download->setDownloadFileName(tempFilePath);
+#endif
+        connect(download, &QWebEngineDownloadItem::finished, [tempFilePath]() {
+            if(!QDesktopServices::openUrl(QUrl::fromLocalFile(tempFilePath)))
+                showInfoBox(gt("error-title"), gt("error-opening-file-with"), KiwixApp::instance()->getMainWindow());
```

Comment:
The temporary file will be left around. Shouldn't we remove it?

Reply:
I don't think so, but should be written in OS temporary directory.

Reply:
QTemporaryFile saves it in the tmp directory anyway so I havent changed anything here.